lib/elixir/lib/file.ex

```diff
@@ -1002,7 +1002,7 @@ defmodule File do
     F.open(IO.chardata_to_string(path), normalize_modes(modes, true))
   end
 
-  def open(path, function) when is_function(function) do
+  def open(path, function) when is_function(function, 1) do
     open(path, [], function)
   end
```

Comment:
this function above is giving trouble, because this feature is not documented anywhere, and it's is meant to be part of `File.open/3` but placed here because of clashing with `open/2`. I guess it is here due to backward compatibility reasons.
To make things more confusing the specs contemplate this case, and I read the specs and I don't get why a function is taken as a list of modes.

Also this functions extends to all other File.open* functions.
If we are planning on deprecating this, should we leave a comment for when it should happen.